trie/database.go

```diff
@@ -85,22 +86,109 @@ type Database struct {
 // cachedNode is all the information we know about a single cached node in the
 // memory database write layer.
 type cachedNode struct {
-	blob     []byte              // Cached data block of the trie node
+	node node   // Cached collapsed trie node, excludes raw
+	raw  []byte // Cached raw data (code), excludes node
```

Comment:
You can make a `rawNode` type and store it in `node` instead.

Reply:
We only use this for storing bytecode. No need to cast it to a node.

Reply:
What I meant is that you can avoid having this field and save 16 bytes per cachedNode if you store the raw data in the node field instead.

Reply:
Ah, I see. Hmm, I'm unsure how that would pan out when we need to differentiate between real nodes and rlp blobs. Might be doable with a type case though.

Reply:
Done. PTAL